src/promptflow-devkit/promptflow/batch/_batch_engine.py

```diff
@@ -456,12 +468,12 @@ async def _exec(
 
         # if the batch runs with errors, we should update the errors to ex
         ex = None
-        if not is_timeout:
+        if not is_timeout and self._executor_proxy.allow_aggregation:
             # execute aggregation nodes
             aggr_exec_result = await self._exec_aggregation(batch_inputs, line_results, run_id)
             # use the execution result to update aggr_result to make sure we can get the aggr_result in _exec_in_task
             self._update_aggr_result(aggr_result, aggr_exec_result)
-        else:
+        elif is_timeout:
```

Comment:
We may reorder the logic here to make it more clear:
```
if is_timeout:
elif self._executor_proxy.allow_aggregation
```

Reply:
Updated